src/applications/danews/add_metadata.py

```diff
@@ -0,0 +1,280 @@
+"""
+Add description to dataset and generate an outline to a datasheet describing the
+dataset.
+
+Dependent on:
+    src/applications/danews/dedupe.py
+
+Authors:
+    Kenneth Enevoldsen
+
+"""
+from datasets import load_from_disk
+from pathlib import Path
+import spacy
+
+
+def word_count(batch):
+    nlp = spacy.blank("da")
+    batch["n_tokens"] = [len(doc) for doc in nlp.pipe(batch["text"])]
+    return batch
+
+
+print("Loading dataset")
+path = Path("/work/hope-infomedia_cleaned/infomedia_2000-2021")
+news = load_from_disk(path)
+
+# TODO: fiks
+print("Add n_tokens column")
+news = news.map(word_count, batched=True, batch_size=1024 * 2 * 2, num_proc=16)
+
+# Add description
+print("Creating metadata")
+dates = news["PublishDate"]
+start_date = min(dates).strftime("%d/%m/%Y")
+end_date = max(dates).strftime("%d/%m/%Y")
+
+print("\tCounting unique sources")
+sources = news["Source"]
+n_sources = len(set(sources))
+
+print("\tExtracting duplicate counts")
+n_tokens_ = news["n_tokens"]
+n_tokens = sum(n_tokens_)
+is_duplicate_ = news["is_duplicate"]
+n_clean_tokens = sum(
+    n_tok for n_tok, is_dup in zip(n_tokens_, is_duplicate_) if is_dup is False
+)
+n_non_duplicates = sum([1 for is_dup in is_duplicate_ if is_dup is False])
+n_duplicates = sum([1 for is_dup in is_duplicate_ if is_dup is True])
+n_docs = len(news)
+print("\tExtracting number of low-quality text")
+passed_quality_filter = news["passed_quality_filter"]
+n_low_quality = sum(1 for is_good in passed_quality_filter if is_good is False)
+
+news.info.description = f"""
```

Comment:
Is this going to be replaced by the markdown file?

Reply:
See comment below